app/assets/javascripts/views/searches/results/results_cross.js

```diff
@@ -26,6 +26,21 @@
       setupModal = true;
     }
 
+    //hash-like object to get the lings colored correctly.
+    //hashes the rows of the cross search by ling-ids
+    //associated with that row.  This allows us to color
+    //lings based on their row in getStyler().
+    var rows_by_ling = {}
+    if (resultsJson) {
+      var json_rows = resultsJson.rows;
```

Comment:
You can use either a `.reduce` function here or a `.forEach` if you find it easier to code:

```js
resultsJson.rows.forEach( function(row){

Reply:
Agree with this too.  added to changes.